pkg/ako-operator/akodeploymentconfig_cluster_mapping_helper.go

```diff
@@ -0,0 +1,174 @@
+// Copyright 2022 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package ako_operator
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	akoov1alpha1 "github.com/vmware-samples/load-balancer-operator-for-kubernetes/api/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	kerrors "k8s.io/apimachinery/pkg/util/errors"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/util/conditions"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ListAkoDeplymentConfigSelectClusters list all clusters enabled current akodeploymentconfig
+func ListAkoDeplymentConfigSelectClusters(
+	ctx context.Context,
+	kclient client.Client,
+	log logr.Logger,
+	obj *akoov1alpha1.AKODeploymentConfig) (*clusterv1.ClusterList, error) {
+	// get all clusters can be selected by this akodeploymentconfig's cluster selector
+	selector, err := metav1.LabelSelectorAsSelector(&obj.Spec.ClusterSelector)
+	if err != nil {
+		return nil, err
+	}
+	var clusters clusterv1.ClusterList
+	if err := kclient.List(ctx, &clusters, []client.ListOption{
+		client.MatchingLabelsSelector{Selector: selector},
+	}...); err != nil {
+		return nil, err
+	}
+	// remove clusters that:
+	// 1. not ready
+	// 2. management cluster
+	// 3. previously selected by other adc objects
+	var newItems []clusterv1.Cluster
+	var allErrs []error
+	for _, cluster := range clusters.Items {
+		if !SkipCluster(&cluster) {
+			// management cluster can't be selected by other adc objects
+			// except the management cluster AKODeploymentConfig
+			if cluster.Namespace == akoov1alpha1.TKGSystemNamespace &&
+				obj.Name != akoov1alpha1.ManagementClusterAkoDeploymentConfig {
```

Comment:
shouldn't it be `==`, when ADC name equals to `install-ako-for-mgmt`, it selects no cluster, (or should it select mgmt cluster only??)

Reply:
the for loop here is to filter those clusters that should **not** be selected by `obj` adc.

- When cluster in `tkg-system` namespace and the adc name equals to `install-ako-for-mgmt` , it should be selected, no need to skip here.
- When the cluster is in `tkg-system` namespace and adc name is not equals to `install-ako-for-mgmt` , it should be skipped.

So, the condition for mc cluster should be skipped like:
```
cluster in tkg-system namespace(indicate cluster is a mc) && adc is not install-ako-for-mc
```

Reply:
install-ako-for-mgmt has the very **specific** cluster selector: `cluster-role.tkg.tanzu.vmware.com/management: ""`, which will only select the management cluster since this label will only exist in the management cluster.